### PR TITLE
Reorganize the code in some functions

### PR DESCRIPTION
**Pull request by DK-DARKmatter**
Reorganize the code in bind(), connect() and shutdown(). Make them
more readable and clear.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -129,17 +129,17 @@ pub extern "C" fn bind(
...
-        _ => return error(libc::EBADFD, -1),
-    }
-
-    if next!(bind(fd, addr, len)) == -1 {
-        return -1;
+    if let Socket::Created = *sock {
+        match next!(bind(fd, addr, len)) {
+            0 => {
+                *sock = Socket::Bound(Parameters { client: false });
+                return 0;
```
Comment (npmccallum) [inline src/lib.rs]: `return 0;` => `0`

File: src/lib.rs [hunk 2]
```diff
@@ -129,17 +129,17 @@ pub extern "C" fn bind(
...
+            },
+            r => r,
+        };
```
Comment (npmccallum) [inline src/lib.rs:139]: Remove semi-colon.

File: src/lib.rs [hunk 3]
```diff
@@ -230,17 +230,17 @@ pub extern "C" fn connect(
...
-        _ => return error(libc::EBADFD, -1),
-    }
-
-    if next!(connect(fd, addr, len)) == -1 {
-        return -1;
+    if let Socket::Created = *sock {
+        match next!(connect(fd, addr, len)) {
+            0 => {
+                *sock = Socket::Connected;
+                return 0;
```
Comment (npmccallum) [inline src/lib.rs]: Same as above.

File: src/lib.rs [hunk 4]
```diff
@@ -317,15 +317,15 @@ pub extern "C" fn shutdown(fd: c_int, how: c_int) -> c_int {
...
-        _ => return error(libc::EBADFD, -1),
-    }
-
-    if next!(shutdown(fd, how)) == -1 {
-        return -1;
+    if let Socket::Established = *sock {
+        match next!(shutdown(fd, how)) {
+            0 => {
+                *sock = Socket::Shutdown;
+                return 0;
```
Comment (npmccallum) [inline src/lib.rs]: Same as above.

**Review by npmccallum — CHANGES_REQUESTED** [review]

File: src/lib.rs (hunk 2 above)
Comment (DK-DARKmatter) [reply src/lib.rs:139]: When I removed the semi-colon, `cargo build` would fail, how to avoid that?

Comment (npmccallum) [reply src/lib.rs:139]: Oops. Keep the semi-colon. Put `return` immediately before `match`.

**Review by npmccallum — CHANGES_REQUESTED** [review]

File: src/lib.rs (hunk 2 above)
Comment (DK-DARKmatter) [reply src/lib.rs:139]: If so, will it be good to change `r => r` to `r => error(libc::EBADFD, -1)`? This makes it even more simple and clear.

Comment (npmccallum) [reply src/lib.rs:139]: No. `r` is the error return value from `bind()`. We want to propagate that upward. We only return `error(libc::EBADFD, -1)` when the socked is in a bad state.

**Review by npmccallum — APPROVED** [review]

**Pull request merged**